### PR TITLE
Use stable URLs for patch-diff GitHub patches

### PR DESCRIPTION
**Pull request by adamjstewart**
@tgamblin we missed a couple in #29239.

### CODE REVIEW

File: var/spack/repos/builtin/packages/trilinos/package.py [hunk 1]
```diff
@@ -391,8 +391,8 @@ class Trilinos(CMakePackage, CudaPackage, ROCmPackage):
     patch('fix_clang_errors_12_18_1.patch', when='@12.18.1%clang')
     patch('cray_secas_12_12_1.patch', when='@12.12.1%cce')
     patch('cray_secas.patch', when='@12.14.1:12%cce')
-    patch('https://patch-diff.githubusercontent.com/raw/trilinos/Trilinos/pull/10545.patch',
-          sha256='7f446d8bdcdc7ec29e1caeb0faf8d9fd85bd470fc52d3a955c144ab14bb16b90',
+    patch('https://patch-diff.githubusercontent.com/raw/trilinos/Trilinos/pull/10545.patch?full_index=1',
+          sha256='62272054f7cc644583c269e692c69f0a26af19e5a5bd262db3ea3de3447b3358',
```
Comment (sethrj) [inline var/spack/repos/builtin/packages/trilinos/package.py:395]: This one's on me 😬

**Review by sethrj — DISMISSED** [review]

File: lib/spack/spack/audit.py [hunk 2]
```diff
@@ -298,7 +298,7 @@ def _check_build_test_callbacks(pkgs, error_cls):
 def _check_patch_urls(pkgs, error_cls):
     """Ensure that patches fetched from GitHub have stable sha256 hashes."""
     github_patch_url_re = (
-        r"^https?://github\.com/.+/.+/(?:commit|pull)/[a-fA-F0-9]*.(?:patch|diff)"
+        r"^https?://(?:patch-diff\.)?github(?:usercontent)?\.com/.+/.+/(?:commit|pull)/[a-fA-F0-9]*.(?:patch|diff)"
```
Comment (sethrj) [inline lib/spack/spack/audit.py]: Line too long 😢

Comment (adamjstewart) [reply lib/spack/spack/audit.py]: I thought flake8 automatically ignored URLs...

Comment (sethrj) [reply lib/spack/spack/audit.py]: Maybe you need a meta url regex? 😂

Comment (adamjstewart) [reply lib/spack/spack/audit.py]: Oh right, there's no way for flake8 to tell that this is supposed to be a URL lol

**Review by wdconinc — APPROVED** [review]

**Review by sethrj — APPROVED** [review]

**Pull request merged**